src/cdk/wizards/previewStateMachineCDKWizard.ts

```diff
@@ -18,62 +18,75 @@ import {
 } from '../../shared/wizards/multiStepWizard'
 import { CdkAppLocation } from '../explorer/cdkProject'
 import { detectCdkProjects } from '../explorer/detectCdkProjects'
-import { ConstructNode } from '../explorer/nodes/constructNode'
 import { AppNode } from '../explorer/nodes/appNode'
+import { ConstructNode } from '../explorer/nodes/constructNode'
 
 export interface CdkAppLocationPickItem {
     label: string,
-    cdkApplocation: CdkAppLocation
+    cdkApplocation: CdkAppLocation | undefined
 }
+
+export interface TopLevelNodePickItem {
+    label: string,
+    topLevelNode: ConstructNode | undefined
+}
+
 export interface ConstructNodePickItem {
     label: string,
-    //stateMachineNode: ConstructNode | PlaceholderNode
-    stateMachineNode: ConstructNode
-    //stateMachineNode: string
+    stateMachineNode: ConstructNode | undefined
 }
 
 interface PreviewStateMachineCDKWizardResponse {
-    cdkApplication: CdkAppLocationPickItem
+    cdkApplication: CdkAppLocationPickItem,
+    topLevelNode: TopLevelNodePickItem,
     stateMachine: ConstructNodePickItem
 }
 
 export default class PreviewStateMachineCDKWizard extends MultiStepWizard<PreviewStateMachineCDKWizardResponse> {
     private cdkApplication?: CdkAppLocationPickItem
+    private topLevelNode?: TopLevelNodePickItem
     private stateMachine?: ConstructNodePickItem
     private promptUser: typeof picker.promptUser
 
     public constructor(promptUser?: typeof picker.promptUser) {
         super()
-
         this.promptUser = promptUser || picker.promptUser.bind(picker)
     }
 
     protected get startStep() {
-        return this.CREATE_TEMPLATE_ACTION
+        return this.SELECT_WORKSPACE_ACTION
     }
 
-    private readonly CREATE_TEMPLATE_ACTION: WizardStep = async () => {
+    private readonly SELECT_WORKSPACE_ACTION: WizardStep = async () => {
         const cdkAppLocations: CdkAppLocation[] = await detectCdkProjects(vscode.workspace.workspaceFolders)
-
-        if (cdkAppLocations.length === 0) return wizardContinue(this.TEMPLATE_FORMAT_ACTION)
-        //need to pick out only the applications containing a state machine 
         const CDK_APPLOCATIONS: CdkAppLocationPickItem[] = []
+
         cdkAppLocations.map(obj => {
```

Comment:
The map function transforms each element, and returns the resulting transformed array. You can replace this with:

```
const CDK_APPLOCATIONS: CdkAppLocationPickItem[] = cdkAppLocations.map(obj => {
            return {
                label: getCDKAppWorkspaceName(obj.cdkJsonPath),
                cdkApplocation: obj
            }
        })
```